File: fixtures/scale/dart/large/lib/ciphers/atbash/atbash_cipher.dart
import '../../core/cipher.dart';
import '../../util/alphabet.dart';
import 'atbash_key.dart';

/// Mirrors the alphabet: A maps to Z, B to Y, and so on.
class AtbashCipher implements Cipher {
  // Atbash is keyless; the key type exists so the suite wiring is uniform.
  AtbashCipher(AtbashKey key);

  @override
  String name() => 'atbash';

  @override
  String encrypt(String plaintext) => _mirror(plaintext);

  @override
  String decrypt(String ciphertext) => _mirror(ciphertext);

  String _mirror(String text) {
    final sb = StringBuffer();
    for (final c in cleanAlphabet(text).split('')) {
      sb.write(charAt(alphabetSize - 1 - indexOf(c)));
    }
    return sb.toString();
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/atbash/AtbashCipher.kt
package cryptobench.ciphers.atbash

import cryptobench.core.Cipher
import cryptobench.util.Alphabet

/** Mirrors the alphabet: A maps to Z, B to Y, and so on. */
class AtbashCipher(@Suppress("UNUSED_PARAMETER") key: AtbashKey) : Cipher {

    override fun name(): String = "atbash"

    override fun encrypt(plaintext: String): String = mirror(plaintext)

    override fun decrypt(ciphertext: String): String = mirror(ciphertext)

    private fun mirror(text: String): String {
        val sb = StringBuilder()
        for (c in Alphabet.clean(text)) {
            sb.append(Alphabet.charAt(Alphabet.SIZE - 1 - Alphabet.indexOf(c)))
        }
        return sb.toString()
    }
}
